### PR TITLE
Add async-graphql to Federation third-party support

### PR DESCRIPTION
**Pull request by sunli829**
`async-graphql` is GraphQL server library implemented by rust, fully supports federation. 😁

Example
==========

```shell
git clone https://github.com/sunli829/async-graphql
cd async-graphql
yarn install --cwd examples/federation
cd examples && ./start-feceration.sh
```

### CODE REVIEW

**Comment by sunli829** [comment]
I have submitted. 😁

File: docs/source/federation/other-servers.md [hunk 1]
```diff
@@ -5,15 +5,16 @@ sidebar_title: Third-party support
...
+| ----------- | ------------- | -------------------------------------------------------------------------------- |
+| Go          | gqlgen        | [gqlgen](https://github.com/99designs/gqlgen/tree/master/plugin/federation)      |
+| Java        | graphql-java  | [federation-jvm](https://github.com/apollographql/federation-jvm)                |
+| Kotlin      | graphql-java  | [graphql-kotlin](https://github.com/ExpediaGroup/graphql-kotlin)                 |
+| PHP         | graphql-php   | [php-graphql-federation](https://github.com/pascaldevink/php-graphql-federation) |
+| PHP         | wp-graphql    | [wp-graphql-federation](https://github.com/wp-graphql/wp-graphql-federation)     |
+| Python      | ariadne       | [ariadne](https://github.com/mirumee/ariadne)                                    |
+| Python      | graphene      | [graphene-federation](https://github.com/preply/graphene-federation)             |
+| Ruby        | graphql-ruby  | [apollo-federation-ruby](https://github.com/Gusto/apollo-federation-ruby)        |
+| Rust        | async-graphql | [async-graphql](https://crates.io/crates/async-graphql)                          |
```
Comment (jhampton) [inline docs/source/federation/other-servers.md:18]: @trevor-scheer Should we request a link to Github, not the Crate?

@sunli829 - Thank you for the PR!  We're taking a look.

Comment (trevor-scheer) [inline docs/source/federation/other-servers.md:18]:
```suggestion
| Rust        | async-graphql | [async-graphql](https://github.com/sunli829/async-graphql)                          |
```

**Comment by sunli829** [comment]
Moved example code to https://github.com/sunli829/async-graphql-examples/tree/master/federation.

😁

**Comment by jhampton** [comment]
@sunli829 Can you accept the recommended changes and resolve conflicts here?

**Comment by sunli829** [comment]
I resolved the conflict.😁

**Pull request merged**